src/lib/converter/factories/declaration.ts

```diff
@@ -45,8 +63,16 @@ export function createDeclaration(context: Context, node: ts.Node, kind: Reflect
 
     // Test whether the node is exported
     let isExported: boolean;
-    if (container.kindOf([ReflectionKind.Module, ReflectionKind.ExternalModule])) {
-        isExported = false; // Don't inherit exported state in modules and namespaces
+    if (container.kindOf([ReflectionKind.Module, ReflectionKind.ExternalModule])
+        || ((considerInheritedExportKinds.indexOf(kind) === -1
+            || node.kind === ts.SyntaxKind.VariableDeclaration || node.kind === ts.SyntaxKind.FunctionDeclaration)
+        && !container.kindOf([ReflectionKind.ObjectLiteral, ReflectionKind.TypeLiteral]))) {
```

Comment:
This logic is confusing, can it be simplified? I wouldn't know how to fix this if there was a problem.

Reply:
@aciccarello 

I am not sure about the meaning of "exported" here.

Considering the following source file that is a real-life-example (comments removed):
```typescript
namespace IduIT.Core.Geometry
{
	export enum GeometryTypeNames
	{
		EmptyGeometry = 0,
		Point = 1,
		MultiPoint = 2,
		LineString = 3,
		MultiLineString = 4,
		Polygon = 5,
		MultiPolygon = 6,
		Circle = 7,
	}
	export namespace GeometryTypeNames
	{
		export type TName = 'EmptyGeometry' | 'Point' | 'MultiPoint' | 'LineString' | 'MultiLineString' | 'Polygon' | 'MultiPolygon' | 'Circle';
		export const $isFlags = false;
		const ___labels: { [index: number]: string } = {
			[GeometryTypeNames.EmptyGeometry]: "Leer",
			[GeometryTypeNames.Point]: "Einzel-Punkt (Point)",
			[GeometryTypeNames.MultiPoint]: "Mehrere Punkte (MultiPoint)",
			[GeometryTypeNames.LineString]: "Linie (LineString)",
			[GeometryTypeNames.MultiLineString]: "Mehrere Linien (MultiLineString)",
			[GeometryTypeNames.Polygon]: "Fläche (Polygon)",
			[GeometryTypeNames.MultiPolygon]: "Mehrere Flächen (MultiPolygon)",
			[GeometryTypeNames.Circle]: "Kreis (Circle)",
		}
		export function getLabel(v: GeometryTypeNames): string
		{
			return ___labels[v] || GeometryTypeNames[v];
		}
	}
}
```

Here everything except the "___labels" object is exported.

But as the enum GeometryTypeNames is merges with the namespace GeometryTypeNames without this PR everything (except direct children of modules) inherit the exported state of its parent (``container.flags.isExported``), so would ___labels. This is wrong.

I don't like this part of code, maybe it could be simplieifed, but it works as far as I can see and consideres the following things:

- if anything could inherit the containers isExported flag, it just could be
  - properties, accessors and methods of a class or interface.
  - signatures and parameters of its function/method
  - index signatures
  - sub properties of an exported type or object literal
- the exported flag can explicitely not be inherited for variables/consts, functions, classes, interfaces, namespaces.
- the "excludeNotExported" option

I think the main issue is, that currently classes and namespaces are merged together to a single reflection model and variables and functions within the namespace are treated as static properties or methods of the class.

See src/lib/converter/nodes/variable.ts
```typescript
const kind = scope.kind & ReflectionKind.ClassOrInterface ? ReflectionKind.Property : ReflectionKind.Variable;
const variable = createDeclaration(context, node, kind, name);
```
I think this should be refactored, keeping variables variables and functions functions.